### PR TITLE
Search products

### PR DESCRIPTION
**Pull request by bajajcodes**
## Search Products Bar
- Search products using search bar and get searched products on search button click and on input in search bar.
- Search bar is available only for products page and nested products pages.
- Search bar is made up of JSX elements.

## Deploy Preview
https://deploy-preview-12--3on3store-react.netlify.app/

### CODE REVIEW

**Review by junaidshaikh-js — APPROVED** [review]
Good work. 

Learned about useSearchParams hook. Clean code and can be merged.

**Review by RakshithBhat03 — APPROVED** [review]
- The code looks clean and well structured. 
- Good use of custom hooks.

File: src/pages/products/main/cardContainer.js [hunk 1]
```diff
@@ -15,15 +16,17 @@ function sastaLoader() {
...
 function CardContainer() {
+  const [getSearchedProducts] = useSearchFeature();
   const { filteredProducts } = useProducts();
+  const products = getSearchedProducts(filteredProducts);
 
   return (
     <div className="products-cards-container">
-      {filteredProducts &&
-        filteredProducts.map((product) => (
+      {products &&
```
Comment (yaswanthmyneni) [inline src/pages/products/main/cardContainer.js:25]: suggestion: check like products?.length > 0 instead of products

**Review by yaswanthmyneni — APPROVED** [review]
code is clean, got to know about useSearchParams, you can merge it.

File: src/pages/products/main/cardContainer.js (hunk 1 above)
Comment (bajajcodes) [reply src/pages/products/main/cardContainer.js:25]: Suggestion Adopted.

**Pull request merged**